### ISSUE TITLE
file.replace does not allow itself to be watched?

### ISSUE THREAD

**Issue by jhgg** [saltstack/salt#21336]
Noticing this issue where the service wouldn't restart when the file was changed by file.replace. 

Didn't see anything in documentation about this not working - can someone clarify? 

SLS file below:

```yaml
nginx:
  pkg:
    - installed

  service:
    - running
    - enable: True
    - reload: True

/etc/nginx/nginx.conf:
  file:
    - replace
    - pattern: '#? server_names_hash_bucket_size (\d+);'
    - repl: 'server_names_hash_bucket_size 128;'
    - watch_in:
      - service: nginx

```

Putting the `watch:` in `nginx.service` gives same error. 

Error On Salt Master:

```
          ID: listener_nginx
    Function: service.mod_watch
        Name: nginx
      Result: False
     Comment: The following requisites were not found:
                                 watch:
                                     file: /etc/nginx/nginx.conf
     Started:
    Duration:
     Changes:
```

Master/Minion version: salt 2014.7.0 (Helium)
Master/Minion OS:
```
#uname -a
Linux 3.13.0-43-generic #72-Ubuntu SMP Mon Dec 8 19:35:06 UTC 2014 x86_64 x86_64 x86_64 GNU/Linux
#cat /etc/issue
Ubuntu 14.04.1 LTS \n \l
```

**Comment by thatch45** [comment]
@jhgg I think that this is already fixed but not in a release yet. the problem was when a listen call would try to eval requisites when it does not need to. From the looks of things you have a listen_in somewhere else referencing the file and it is colliding with the watch.
if you change that listen_in to a watch_in the problem should go away until we can get the fix out in a release

**Comment by rallytime** [comment]
@jhgg If you indeed have some `listen` or `listen_in` requisites elsewhere in your state, you're hitting this same issue as reported in #20863. This was fixed in #20881 and as stated earlier will be in the next release of salt.

**Comment by cachedout** [comment]
Since we have had no reply here and the issue appears to be fixed, I am going to go ahead and close this issue. @jhgg if it is not working for you on the next release or in the develop branch, please leave a comment on this issue and we will happily re-open it.
